pipelines.yml

```diff
@@ -0,0 +1,17 @@
+- pipeline.id: logs
+  config.string: |
+      input {
+          tcp {
+              port => 4560
+              codec => json_lines
+          }
+      }
+      output {
+          pipeline {
+              send_to => ["nagios", "elasticsearch"]
+          }
+      }
+- pipeline.id: nagios
```

Comment:
Curious... why is one pipeline defined on conf file and another one as `config.string` in `pipelines.yaml`? I don't care much, but muh consistency.. Doesn't the `pipeline.id` default so name of the file?

Reply:
That's consistency for the sake of consistency.  The string pipeline lives on a meta level with respect to other pipelines, so it make sense to put it a level above in the configuration as well so that it's visible what's going on by glancing at the `pipelines.yml` file without having to view individual files and piece together what they collectively do in the reader's mind.

As for the other question, I don't think so.